### PR TITLE
Default to using extension of the file for editor replacement

### LINKED ISSUE #8479: SQLPLAN files no longer open with custom editor
- Azure Data Studio Version: `master`

Steps to Reproduce:

1. Save a query plan to a .SQLPLAN file.  (e.g. Run a query with the Explain option, click the XML link, save the XML file as file.sqlplan).
2. Open the SQLPLAN file

Expected: The plan is rendered in the SQL PLAN custom editor

Actual: XML opens in a query editor document.  This is a regression from 1.13.1.

image

### PR DESCRIPTION
**Pull request by anthonydresser**
This PR fixes #8479

### CODE REVIEW

File: src/sql/workbench/common/editorReplacerContribution.ts [hunk 1]
```diff
@@ -55,6 +56,11 @@ export class EditorReplacementContribution implements IWorkbenchContribution {
 			language = this.modeService.getModeIdByFilepathOrFirstLine(editor.getResource());
 		}
 
+		if (!language) {
+			// just use the extension
+			language = path.extname(editor.getResource().toString()).slice(1); // remove the .
```
Comment (kburtram) [inline src/sql/workbench/common/editorReplacerContribution.ts]: how does this work if the file doesn't have an extension?

Comment (anthonydresser) [reply src/sql/workbench/common/editorReplacerContribution.ts]: I'll double check error cases but extname should be and empty string so slice will return and empty string.

File: src/sql/workbench/common/editorReplacerContribution.ts [hunk 2]
```diff
@@ -55,6 +56,11 @@ export class EditorReplacementContribution implements IWorkbenchContribution {
 			language = this.modeService.getModeIdByFilepathOrFirstLine(editor.getResource());
 		}
 
+		if (!language) {
```
Comment (chlafreniere) [inline src/sql/workbench/common/editorReplacerContribution.ts]: can you replace an editor with another untitled editor?

Comment (anthonydresser) [reply src/sql/workbench/common/editorReplacerContribution.ts]: What do you mean?

File: src/sql/workbench/common/editorReplacerContribution.ts (hunk 1 above)
Comment (anthonydresser) [reply src/sql/workbench/common/editorReplacerContribution.ts]: I've verified that the extname method returns an empty string if the string has no extname, then slice(1) will just return an empty string again.

**Pull request merged**